implants/lib/eldritch/src/assets/read_binary_impl.rs

```diff
@@ -1,6 +1,31 @@
-use anyhow::Result;
+use std::sync::mpsc::{channel, Receiver};
 
-pub fn read_binary(src: String) -> Result<Vec<u32>> {
+use anyhow::{Context, Result};
+use pb::c2::FetchAssetResponse;
+use starlark::{eval::Evaluator, values::list::ListRef};
+
+use crate::runtime::{messages::FetchAssetMessage, Environment};
+
+fn read_binary_remote(rx: Receiver<FetchAssetResponse>) -> Result<Vec<u32>> {
+    // Wait for our first chunk
+    let resp = rx.recv()?;
+
+    let mut res: Vec<u32> = vec![];
+
+    let mut first_chunk = resp.chunk.iter().map(|x| *x as u32).collect::<Vec<u32>>();
+    res.append(&mut first_chunk);
+
+    // Listen for more chunks and write them
+    for resp in rx {
+        let mut new_chunk = resp.chunk.iter().map(|x| *x as u32).collect::<Vec<u32>>();
+
+        res.append(&mut new_chunk);
+    }
+
+    Ok(res)
+}
+
+fn read_binary_local(src: String) -> Result<Vec<u32>> {
```

Comment:
i think maybe embedded might be less ambiguous than local? That being said, im sure all the file functions needs some sort of adjustment lol.